### PR TITLE
Feature/advanced search incorporation date

### PR DESCRIPTION
**Pull request by ethatcher83**
Added incorporation date fields to search and results page
Added validation for results page incorporation dates

Resolves: BI-8914 BI-8928 BI-9336 BI-9337 BI-9362 BI-9376 BI-9382

### CODE REVIEW

File: src/controllers/advanced-search/search.controller.ts [hunk 1]
```diff
@@ -1,12 +1,13 @@
-import { Request, Response } from "express";
+import { NextFunction, Request, Response } from "express";
```
Comment (caismith-ch) [inline src/controllers/advanced-search/search.controller.ts:1]: I can't see this used anywhere in the class, is this required?

Comment (ethatcher83) [reply src/controllers/advanced-search/search.controller.ts:1]: Not being used, will remove it

File: src/controllers/utils/advanced-search-validation.ts [hunk 2]
```diff
@@ -0,0 +1,94 @@
+import { check, param } from "express-validator";
```
Comment (caismith-ch) [inline src/controllers/utils/advanced-search-validation.ts:1]: Is this `param` used also. I can see `error.param` but i'm not sure whether that is the same thing?

File: src/controllers/utils/advanced-search-validation.ts [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+function checkStringNotNullOrEmpty (checkString: string) {
+    return checkString !== null && checkString !== undefined && checkString.length > 0;
+}
+
+function checkIfDateInFuture (date: string) {
+    const dateMoment = moment(date, "DD/MM/YYYY");
+    const checkDate = dateMoment.toDate();
+    const now = new Date();
+    return now < checkDate;
+}
```
Comment (caismith-ch) [inline src/controllers/utils/advanced-search-validation.ts:94]: I think adding the return type `: boolean` to these will make it easier at a quick glance to see what these functions do. Also another common standard for booleans is to use `is` as a prefix, for example:
`isDateBeforeInitial`
`isStringNotNullOrEmpty`
`isDateInFuture`

Comment (hwilliams1) [reply src/controllers/utils/advanced-search-validation.ts]: Changed as per Cai's comments

File: src/controllers/utils/advanced-search-validation.ts (hunk 2 above)
Comment (hwilliams1) [reply src/controllers/utils/advanced-search-validation.ts]: Unused import removed

**Review by caismith-ch — APPROVED** [review]

**Pull request merged**